mlflow/langchain/utils.py

```diff
@@ -314,8 +314,14 @@ def _validate_and_prepare_lc_model_or_path(lc_model, loader_fn, temp_dir=None):
             type(lc_model.llm).__name__,
         )
 
-    if isinstance(lc_model, langchain.agents.agent.AgentExecutor) and not any(
-        isinstance(lc_model.agent.llm_chain.llm, supported_llm) for supported_llm in _SUPPORTED_LLMS
+    if (
```

Comment:
nit: Can we extract this condition as a self-explanatory function like `is_agent_contain_unsupported_llm(chain)`?

Reply:
Emm it's only used here and we'll need to extract _SUPPORTED_LLMS if moving out, maybe it's fine keeping here and add a comment?

Reply:
Updated!